### PR TITLE
Fix: Restore pgr_boyerMyrvold to build and fix compilation bugs

### PR DESCRIPTION
**Pull request by Mohit242-bit**
Problem:
`pgr_boyerMyrvold` was developed  but was never functional due to several bugs that prevented it from compiling. The source files existed but were excluded from the build system in `src/planar/CMakeLists.txt` and `sql/planar/CMakeLists.txt`.

Changes proposed in this pull request:
| File | Bug | Fix |
|------|-----|-----|
| `boyerMyrvold_driver.cpp` | Stale includes with `pgr_` prefix | Updated to `alloc.hpp`, `assert.hpp` etc. |
| `boyerMyrvold_driver.h` | Function name mismatch | Renamed to `pgr_do_boyerMyrvold` |
| `boyerMyrvold_driver.cpp` | Incorrect `UndirectedGraph` constructor | Changed to default constructor |
| `boyerMyrvold.c` | Outdated `IID_t_rt` members | Changed `.source/.target` to `.from_vid/.to_vid` |
| `boyerMyrvold.hpp` | Unused header | Removed `is_kuratowski_subgraph.hpp` |
| `CMakeLists.txt` (src/sql) | Files excluded from build | Added source and SQL files back to build system |

## Testing
- Clean build from scratch with 0 warnings.
- Verified function returns correct planar embedding (18 edges on sample graph).
- K₅ (non-planar graph) correctly returns 0 rows.
- Added 4 new pgTAP test files following existing `isPlanar` patterns.
- All 153 pgTAP tests pass 

Note:
RST documentation is not included in this PR — please let me know if it should be added here or handled in a follow-up.
 
@pgRouting/admins

## Summary by CodeRabbit

* **New Features**
  * Added pgr_boyermyrvold(text) for Boyer–Myrvold planar embedding; returns seq, source, target, cost (available in v4.1).

* **Tests**
  * Added comprehensive pgTAP tests: edge cases, type/signature checks, inner-query behavior, no-crash verification, and version gating.

* **Documentation**
  * Added user documentation and localized messages for the new planar routine; listed in experimental functions.

* **Packaging**
  * Included new SQL files so the routine is packaged and installed.

### CODE REVIEW

**Review by cvvergara — CHANGES_REQUESTED** [review]
@Mohit242-bit 
Documentation is missing we need documentation
Looks like there was no documentation created when this was attempted on 2020
Please add documentation.
you will need to add the `.rst` of the function under doc/planar  directory
also modify the doc/_static/page_history.js
It is an experimental function.

Also the SQL files should have the correct version 4.1
For example
```
--v3.2
CREATE FUNCTION  ....
```
Change to
```
--v4.1
CREATE FUNCTION 
```

File: doc/planar/pgr_boyerMyrvold.rst [hunk 1]
```diff
@@ -0,0 +1,135 @@
+:file: This file is part of the pgRouting project.
+:copyright: Copyright (c) 2020-2026 pgRouting developers
```
Comment (cvvergara) [inline doc/planar/pgr_boyerMyrvold.rst:2]: Actually the file is originally from 2026 so 
```
:copyright: Copyright (c) 2026-2026 pgRouting developers
```

File: doc/planar/pgr_boyerMyrvold.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+* Applicable only for **undirected** graphs.
+* The algorithm does not consider traversal costs in the calculations.
+* Running time: :math:`O(|V|)`
+
+|Boost| Boost Graph Inside
+
+Signatures
+-------------------------------------------------------------------------------
+
+.. rubric:: Summary
```
Comment (cvvergara) [inline doc/planar/pgr_boyerMyrvold.rst:50]: There is only one function so no summary word is needed

**Review by cvvergara — CHANGES_REQUESTED** [review]

File: docqueries/planar/boyerMyrvold.result [hunk 3]
```diff
@@ -5,66 +5,100 @@ SET
...
+                            'SELECT id, source, target, cost, reverse_cost FROM edges ')
                         WHERE component = 14)
                        )
 $$
  );
- seq | source | target | cost
------+--------+--------+------
-   1 |     14 |     15 |    1
-(1 row)
+ERROR:  No edges found
```
Comment (cvvergara) [inline docqueries/planar/boyerMyrvold.result:88]: hmmm, the inner query is not returning edges

```
SELECT node FROM pgr_connectedComponents(
                            'SELECT id, source, target, cost, reverse_cost FROM edges ')
                        WHERE component = 14;
 node 
------
(0 rows)
```
but this one works:
```
SELECT node FROM pgr_connectedComponents(
                            'SELECT id, source, target, cost, reverse_cost FROM edges ')
                        WHERE component = 13;
 node 
------
   13
   14
(2 rows)
```

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Comment by cvvergara** [comment]
Because of the unused include I went to read the boost documentation for [boyer_myrvold](https://www.boost.org/doc/libs/latest/libs/graph/doc/boyer_myrvold.html)

Computing a planar embedding for a graph if it is planar, otherwise finding a set of edges that forms an obstructing Kuratowski subgraph: 
```
if (boyer_myrvold_planarity_test(boyer_myrvold_params::graph = g,
                                 boyer_myrvold_params::embedding = embedding_pmap,
                                 boyer_myrvold_params::kuratowski_subgraph = out_itr
                                 )
    )
{
  //do something with the embedding in embedding_pmap
}
else
{
  //do something with the kuratowski subgraph output to out_itr
}
```

Right now when the graph is not planar its giving an empty result, the intention is to output the kuratowski_subgraph when its not planar.

See this [example](https://www.boost.org/doc/libs/latest/libs/graph/example/kuratowski_subgraph.cpp) from boost
maybe it can be used.

**Comment by cvvergara** [comment]
Hi @Mohit242-bit 

Was looking at this:
it was done by GSoC student on [2020](https://github.com/pgRouting/pgrouting/wiki/GSoC-2020-Implement-Boyer-Myrvold-Planarity-Testing-and-Make-Connected-in-pgRouting) 

The thing here is:
Boyer Myrvold Planarity Testing: the resulting function was [pgr_isPlanar](https://docs.pgrouting.org/latest/en/pgr_isPlanar.html)

[boost documentation](https://www.boost.org/doc/libs/latest/libs/graph/doc/boyer_myrvold.html)

> As a side-effect of this test, a planar embedding can be constructed if the graph is planar or a minimal set of edges that form a [Kuratowski subgraph](https://www.boost.org/doc/libs/latest/libs/graph/doc/planar_graphs.html#kuratowskisubgraphs) can be found if the graph is not planar.

So I think it is my bad, because he was trying to use the side effect to create a new function which was not on the proposal. And I let the code slide .... Thing that I am not doing anymore  see #2963 which is a PR that will not get merged, (in this case because a problem about not ending and not having a visitor that potentially could allow us to end the process)

I am not letting this kind of code slide any more.
What I think is best is to:
Open a ticket about dead code boyer... 
And remove the code
